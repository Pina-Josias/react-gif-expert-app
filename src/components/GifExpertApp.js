import React, { useState } from "react";
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = ({ defaultCategory = [] }) => {
  const [categories, setCategories] = useState(defaultCategory);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((element) => (
          <GifGrid key={element} category={element} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
