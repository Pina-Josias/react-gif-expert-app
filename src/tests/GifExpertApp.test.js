import GifExpertApp from "../components/GifExpertApp";
import React from "react";
import { shallow } from "enzyme";

describe("Pruebas sobre el documento de <GifExpertApp/>", () => {
  test("debe hacer match con el snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
  });
});
