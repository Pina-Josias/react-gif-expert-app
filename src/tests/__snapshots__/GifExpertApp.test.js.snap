// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el documento de <GifExpertApp/> debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre el documento de <GifExpertApp/> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
